frame/election-provider-support/src/lib.rs

```diff
@@ -372,24 +372,47 @@ pub trait ElectionProvider {
 
 	/// Indicate if this election provider is currently ongoing an asynchronous election or not.
 	fn ongoing() -> bool;
+}
 
-	/// Elect a new set of winners, without specifying any bounds on the amount of data fetched from
-	/// [`Self::DataProvider`]. An implementation could nonetheless impose its own custom limits.
+pub trait BoundedElectionProvider: ElectionProviderBase {
+	/// The upper bound on election winners.
+	type MaxWinners: Get<u32>;
+
+	/// Elect a new set of winners, bounded by `MaxWinners` but an unbounded
+	/// number of backers per winner. Data is fetched from
+	/// [`ElectionProviderBase::DataProvider`]. An implementation could
+	/// nonetheless impose its own custom limits.
 	///
-	/// The result is returned in a target major format, namely as *vector of supports*.
+	/// The result is returned in a target major format, namely as *Bounded vector of
+	/// support*.
 	///
-	/// This should be implemented as a self-weighing function. The implementor should register its
-	/// appropriate weight at the end of execution with the system pallet directly.
+	/// This should be implemented as a self-weighing function. The implementor
+	/// should register its appropriate weight at the end of execution with the
+	/// system pallet directly.
+	fn elect() -> Result<BoundedSupports<Self::AccountId, Self::MaxWinners>, Self::Error>;
+}
+
+pub trait ElectionProvider: ElectionProviderBase {
+	/// Elect a new set of winners, without specifying any bounds on the amount
+	/// of data fetched from [`ElectionProviderBase::DataProvider`].
+	/// An implementation could nonetheless impose its own custom limits.
+	///
+	/// The result is returned in a target major format, namely as *vector of
+	/// support*.
+	///
+	/// This should be implemented as a self-weighing function. The implementor
+	/// should register its appropriate weight at the end of execution with the
+	/// system pallet directly.
```

Comment:
```suggestion
	/// Same a [`BoundedElectionProvider`], but no bounds are imposed on the number of winners.
```

To avoid duplicate facts.

Also, I would move both of these docs to the trait, and leave the function to have a very small, rather rudimentary document like:

```
/// Perform the actual election process.
fn elect() -> ..
```